docs/designs/CodeFlare-SDK-design-doc.md

```diff
@@ -0,0 +1,141 @@
+# CodeFlare SDK Design Document
+
+Team: Distributed Workloads - Orchestration
+
+## Introduction
+This document outlines the design of the Project CodeFlare SDK, a Python SDK that facilitates interactions between users and the distributed workloads component of Red Hat OpenShift AI(RHOAI)/ OpenDataHub(ODH). Users, in this instance, are both data scientists and MLOps Engineers. The SDK provides a high-level abstraction for managing machine learning(ML) workflows, jobs and distributed computing resources.
+
+This document should be followed when adding new functionality along with being updated as the design evolves.
+
+## Objectives
+1. Provide a Pythonic, user-friendly interface for interaction with distributed workloads components.
+2. Integrate with Kueue for managing workload queues, quota management, job orchestration and scheduling.
+3. Provide separate functionalities for data scientists/ MLops engineers and platform administrators.
+4. Allow data scientists to manage ML workflows, workloads and jobs.
+5. Leverage Ray and [Kubeflow][Kubeflow] for workloads and jobs.
```

Comment:
I think you are missing a valid url here

Reply:
The URLs are all stacked at the end of document to make it easier for reading MD without URLs distracting (https://github.com/project-codeflare/codeflare-sdk/pull/703/files#diff-387eb08ad63e5ef0159223fefdf1359e9ae0573ddecfd3e2032fdb4e35275c9fR131-R141) 😅